flink-annotations/src/main/java/org/apache/flink/FlinkVersion.java

```diff
@@ -69,9 +71,14 @@ public boolean isNewerVersionThan(FlinkVersion otherVersion) {
     }
 
     /** Returns all versions equal to or higher than the selected version. */
-    public Set<FlinkVersion> orHigher() {
+    public Set<FlinkVersion> orHigherMinusSnapshot() {
```

Comment:
Suggestions are totally welcomed! :D

Reply:
`orHigherExcludingLatest`? 

I feel like "Snapshot" is not really correct here; I would expect e.g. `release-1.15` to contain 1.15 as the highest one as well (and not 1.16). But the question is whether 1.15 should be excluded on `release-1.15` or not (no idea, I don't know the intention).

Flagging the releases (`boolean released`). would be another option.

Reply:
@twalthr @zentol ?

Reply:
If anything the enum should have a boolean flag, but it seems somewhat overly complicated and isn't really intuitive.

For example, let's say we add this with the flag to 1.15. In the 1.15.0 release the `1_15` enum would still be marked as "not released", because if it were marked as released then the compatibility tests would complain. So we have release that considers itself as not released.

We also end up with _additional_ manual steps on _both_ the `master` and `release-1.15` branch, as they have to set this flag to true after 1.15.0 is released. But only after someone actually went through the trouble of creating the snapshots for the compatibility tests. And if we add additional table api compatibility tests, then this step just got worse.

I'm wondering if we shouldn't separate this distinction from the enum itself.

For example, there could be a flag (or separate ones per test) in the compatibility tests for whether they should also cover the latest version.
Or they could just ignore versions for which the files don't exist. Because as is we could never drop the 1.3 snapshots, because the enum still claims 1.3 exists.

Reply:
@zentol After a discussion with @twalthr we came up with another idea.
Keep only one enum the `FlinkVersion` without flags, and change the method to a `rangeOf` so that the tests can still define what they need. Additionally, if you agree with that, we will just change the steps `14` & `15` in the `Checklist to declare the process completed` section so that the release manager needs only to adjust the `CURRENT_VERSION` on `TypeSerializerUpgradeTestBase`.

Pushed the commit, so please check and let me know what do you think.